test/NuGet.Core.Tests/NuGet.PackageManagement.Test/BuildIntegratedTests.cs

```diff
@@ -695,386 +696,441 @@ public async Task InstallMultiplePackage()
 
-                await nuGetPackageManager.InstallPackageAsync(buildIntegratedProject, versioning105, new ResolutionContext(), new TestNuGetProjectContext(),
-                        sourceRepositoryProvider.GetRepositories(), sourceRepositoryProvider.GetRepositories(), CancellationToken.None);
+            var message = string.Empty;
 
-                // Act
-                await nuGetPackageManager.InstallPackageAsync(buildIntegratedProject, versioning107, new ResolutionContext(), new TestNuGetProjectContext(),
-                        sourceRepositoryProvider.GetRepositories(), sourceRepositoryProvider.GetRepositories(), CancellationToken.None);
+            await nuGetPackageManager.InstallPackageAsync(buildIntegratedProject, versioning105, new ResolutionContext(), new TestNuGetProjectContext(),
+                    sourceRepositoryProvider.GetRepositories(), sourceRepositoryProvider.GetRepositories(), CancellationToken.None);
 
-                var installedPackages = await buildIntegratedProject.GetInstalledPackagesAsync(CancellationToken.None);
-                var lockFile = ProjectJsonPathUtilities.GetLockFilePath(buildIntegratedProject.JsonConfigPath);
+            // Act
+            await nuGetPackageManager.InstallPackageAsync(buildIntegratedProject, versioning107, new ResolutionContext(), new TestNuGetProjectContext(),
+                    sourceRepositoryProvider.GetRepositories(), sourceRepositoryProvider.GetRepositories(), CancellationToken.None);
 
-                // Assert
-                Assert.Equal(1, installedPackages.Count());
-                Assert.Equal(versioning107, installedPackages.Single().PackageIdentity);
-                Assert.True(File.Exists(lockFile));
-            }
+            // Assert
+            var installedPackages = await buildIntegratedProject.GetInstalledPackagesAsync(CancellationToken.None);
+            string assetsFile = await buildIntegratedProject.GetAssetsFilePathAsync();
+            File.Exists(assetsFile).Should().BeTrue();
+
+            installedPackages.Count().Should().Be(1);
+            installedPackages.Single().PackageIdentity.Should().Be(versioning107);
         }
 
-        [Fact(Skip = "https://github.com/NuGet/Home/issues/10212")]
+        [Fact]
         public async Task UpdatePackageToHighest()
         {
             // Arrange
+            using var pathContext = new SimpleTestPathContext();
+            using var testSolutionManager = new TestSolutionManager(pathContext);
+
+            var projectName = "TestProjectName";
             var nugetVersioningId = "Nuget.Versioning";
-            var oldJson = new PackageIdentity(nugetVersioningId, NuGetVersion.Parse("3.5.0"));
+            var originalInstalledPackage = new PackageIdentity(nugetVersioningId, NuGetVersion.Parse("3.5.0"));
             var sourceRepositoryProvider = TestSourceRepositoryUtility.CreateV2OnlySourceRepositoryProvider();
```

Comment:
Similar feedback here. 
The test  will be more reliable if it doesn't depend on nuget.org

Reply:
I actually see some of the tests in this PR already have the online to local conversion, so that's great.